pinc/bootstrap.inc

```diff
@@ -0,0 +1,76 @@
+<?php
+// This file sets up the common infrastructure for the site. It is used for
+// both HTML pages (via base.inc) and suitable for API calls as well.
+// This file should only include functionality that is common in both.
+//
+// This file sets up the vital infrastructure for the site, including:
+// * including site_vars.php to set global variables
+// * registering uncaught exception handler
+// * connecting to the database
+// * pulling in gettext functions
+//
+// The order of the include_once()s in this file is important!
+
+// Override $relPath to be the absolute path to the directory that
+// contains this file. This allows registered shutdown functions
+// to know where to find files (working directory of the script
+// can change inside the shutdown function based on PHP docs).
+// This also helps things like APC which can bypass stat calls
+// on includes that use absolute paths.
+$relPath=dirname(__FILE__)."/";
+
+// store the current time to calculate total page build time
+$PAGE_START_TIME = microtime(true);
+
+include_once($relPath.'site_vars.php');
+
+// Register autoloader
+spl_autoload_register('dp_class_autoloader');
+
+// Exception handlers are defined differently for HTML and the API
+if($testing)
+{
+    set_exception_handler('test_exception_handler');
+}
+else
+{
+    set_exception_handler('production_exception_handler');
+}
```

Comment:
A nit (feel free to ignore), but would this be less verbose as
```
set_exception_handler($testing ? 'test_exception_handler' : 'production_exception_handler');
```

Reply:
Very nice - thanks!